src/main/java/org/springframework/data/redis/connection/jedis/JedisStreamCommands.java

```diff
@@ -0,0 +1,496 @@
+				JedisConverters.toBytes(groupName), JedisConverters.toBytes(readOffset.getOffset()), mkStream);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xGroupDelConsumer(byte[], org.springframework.data.redis.connection.RedisStreamCommands.Consumer)
+	 */
+	@Override
+	public Boolean xGroupDelConsumer(byte[] key, Consumer consumer) {
+		Assert.notNull(key, "Key must not be null!");
+		Assert.notNull(consumer, "Consumer must not be null!");
+
+		return connection.invoke().from(BinaryJedis::xgroupDelConsumer, MultiKeyPipelineBase::xgroupDelConsumer, key,
+				JedisConverters.toBytes(consumer.getGroup()), JedisConverters.toBytes(consumer.getName())).get(r -> r > 0);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xGroupDestroy(byte[], java.lang.String)
+	 */
+	@Override
+	public Boolean xGroupDestroy(byte[] key, String groupName) {
+		Assert.notNull(key, "Key must not be null!");
+		Assert.hasText(groupName, "Group name must not be null or empty!");
+
+		return connection.invoke()
+				.from(BinaryJedis::xgroupDestroy, MultiKeyPipelineBase::xgroupDestroy, key, JedisConverters.toBytes(groupName))
+				.get(r -> r > 0);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xInfo(byte[])
+	 */
+	@Override
+	public StreamInfo.XInfoStream xInfo(byte[] key) {
+		Assert.notNull(key, "Key must not be null!");
+
+		return connection.invoke().from(BinaryJedis::xinfoStream, (k, r) -> {
+			throw new UnsupportedOperationException("'XINFO' cannot be called in pipeline / transaction mode.");
+		}, key).get(streamInfo -> StreamInfo.XInfoStream.fromList(mapToList(streamInfo.getStreamInfo())));
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xInfoGroups(byte[])
+	 */
+	@Override
+	public StreamInfo.XInfoGroups xInfoGroups(byte[] key) {
+		Assert.notNull(key, "Key must not be null!");
+
+		return connection.invoke().from(BinaryJedis::xinfoGroup, (k, r) -> {
+			throw new UnsupportedOperationException("'XINFO GROUPS' cannot be called in pipeline / transaction mode.");
+		}, key).get(streamGroupInfos -> {
+			List<Object> sources = new ArrayList<>();
+			streamGroupInfos.forEach(streamGroupInfo -> sources.add(mapToList(streamGroupInfo.getGroupInfo())));
+			return StreamInfo.XInfoGroups.fromList(sources);
+		});
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xInfoConsumers(byte[], java.lang.String)
+	 */
+	@Override
+	public StreamInfo.XInfoConsumers xInfoConsumers(byte[] key, String groupName) {
+		Assert.notNull(key, "Key must not be null!");
+		Assert.hasText(groupName, "Group name must not be null or empty!");
+
+		return connection.invoke().from(BinaryJedis::xinfoConsumers, (k, v, r) -> {
+			throw new UnsupportedOperationException("'XINFO CONSUMERS' cannot be called in pipeline / transaction mode.");
```

Comment:
We typically add that guard beforehand, see https://github.com/spring-projects/spring-data-redis/blob/53ce84851cbb0350f5df1daf40d5623d5c7e756f/src/main/java/org/springframework/data/redis/connection/jedis/JedisServerCommands.java#L210-L212

Reply:
Got it. Thanks.